[Storage] Fix #16872: az storage blob now (2.19) requires login even if connection-string is provided

Juliehzl opened the pull request:
**Description**<!--Mandatory-->
<!--Why this PR? What is changed? What is the effect? etc. A high-quality description can accelerate the review process.-->
Fix #16872 

**Testing Guide**
<!--Example commands with explanations.-->
`az account clear`
`az storage blob sync` with account key or connection string 

**History Notes**
<!--If your PR is not customer-facing, use {Component Name} in the PR title. Otherwise, use [Component Name] to allow our pipeline to add the title as a history note. If you need multiple history notes or would like to overwrite the note from the PR title, please fill in the following templates.-->

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

yonzhan commented:
Storage

evelyn-ys reviewed (commented):
Do `azcopy.remove` have the same issue? I remember in #16519 both `remove` and `sync` are modified due to permissions. https://github.com/Azure/azure-cli/blob/bc4d70c7b98065c8132fda465d549da0ba924d9d/src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py#L57-L60

Juliehzl commented:
> Do `azcopy.remove` have the same issue? I remember in #16519 both `remove` and `sync` are modified due to permissions.
> 
> https://github.com/Azure/azure-cli/blob/bc4d70c7b98065c8132fda465d549da0ba924d9d/src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py#L57-L60

because the fix 

> Do `azcopy.remove` have the same issue? I remember in #16519 both `remove` and `sync` are modified due to permissions.
> 
> https://github.com/Azure/azure-cli/blob/bc4d70c7b98065c8132fda465d549da0ba924d9d/src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py#L57-L60

Good point! updated and added test.
